### PR TITLE
Fix SQL formatter issues with ORDER BY/GROUP BY clauses and improve block indentation handling

### LINKED ISSUE #428: Incorrect Line Break in ORDER BY
**Describe the bug**
When using ORDER BY inside a function parameter (e.g., within OVER(...)), the keyword group is incorrectly line-broken.

**Example**
```sql
SELECT e.id
       , e.number
       , ROW_NUMBER() OVER(
 ORDER BY e.register_time DESC) AS oe
  FROM

```

**Expected behavior**
The ORDER BY keyword inside the function parameter should remain on the same line, without introducing an unnecessary line break.
```sql
SELECT e.id
       , e.number
       , ROW_NUMBER() OVER(ORDER BY e.register_time DESC) AS oe
  FROM
```

**Desktop :**
 - IDEA Version : 2024.3.6
 - Doma Tools Version :2.1.2

### PR DESCRIPTION
**Pull request by xterao**
## Description

This PR addresses issue #428 by fixing multiple SQL formatting issues related to ORDER BY and GROUP BY clauses, particularly when used with Doma condition directives and within function parameters.

### 1. Improved ORDER BY formatting with condition directives
- Fixed indentation for ORDER BY clauses wrapped in `/*%if*/` condition directives
- Properly handles ORDER BY within window functions
- Ensures correct alignment when ORDER BY appears after condition directives

### 2. Fixed GROUP BY formatting issues
- Resolved spacing issues between keyword groups and parameter groups
- Fixed indentation calculation for consecutive operators
- Improved handling of WITHIN GROUP clauses for aggregate functions

### 3. Enhanced block indentation logic
- Refactored parent group assignment logic for better consistency
- Fixed indentation calculation for nested conditional blocks
- Improved handling of function parameters containing SQL keywords

### 4. Added comprehensive test coverage
- Added test case for ORDER BY with condition directives
- Added test case for function keywords in condition directives
- Added test case for function names used as column names

### Before:
```sql
SELECT e.id
       , ROW_NUMBER() /*%if order */OVER(ORDER BY e.manager_id DESC)/*%end*/ AS row_num
       , RANK() OVER(PARTITION BY department_id
/*%if order */ORDER BY e.manager_id DESC/*%end */
ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) AS rank_num
```

### After:
```sql
SELECT e.id
       , ROW_NUMBER()
         /*%if order */
         OVER(ORDER BY e.manager_id DESC)
         /*%end*/
         AS row_num
       , RANK() OVER(PARTITION BY department_id
                     /*%if order */
                     ORDER BY e.manager_id DESC
                     /*%end */
                     ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) AS rank_num
```

## Testing

All existing tests pass, and new test cases have been added to ensure the formatting improvements work correctly with:
- Window functions with ORDER BY clauses
- Conditional directives wrapping ORDER BY/GROUP BY
- Complex nested SQL expressions
- Aggregate functions with WITHIN GROUP clauses

## Related Issues

Fixes #428

## Checklist

- [x] Code follows the project's style guidelines
- [x] Tests have been added to cover the changes
- [x] All existing tests pass
- [x] The formatter correctly handles edge cases
- [x] Documentation has been updated if necessary

### CODE REVIEW

File: src/main/kotlin/org/domaframework/doma/intellij/formatter/block/word/SqlFunctionGroupBlock.kt [hunk 1]
```diff
@@ -38,7 +39,7 @@ class SqlFunctionGroupBlock(
         indent.groupIndentLen = createGroupIndentLen()
     }
 
-    override fun createBlockIndentLen(): Int = parentBlock?.indent?.groupIndentLen ?: 0
+    override fun createBlockIndentLen(): Int = parentBlock?.indent?.groupIndentLen?.plus(1) ?: 0
```
[the remark this answers is not in the document]
Comment (xterao) [reply src/main/kotlin/org/domaframework/doma/intellij/formatter/block/word/SqlFunctionGroupBlock.kt:42]: For function name indentation, it needs to be calculated as the parent group’s indent (up to the end of the parent’s text range) plus one space.
If the function is placed directly under a conditional loop directive, using only the parent group’s indent results in being one space short.

Indentation for elements after a line break follows the same rule—parent group indent +1.

File: src/main/kotlin/org/domaframework/doma/intellij/formatter/util/SqlKeywordUtil.kt [hunk 2]
```diff
@@ -313,16 +319,17 @@ class SqlKeywordUtil {
                 "join" to setOf("outer", "inner", "left", "right"),
                 "outer" to setOf("left", "right"),
                 "inner" to setOf("left", "right"),
-                "by" to setOf("group", "order", "first"),
-                "and" to setOf("between"),
+                "group" to setOf("within"),
+                "by" to setOf("group", "order", "first", "partition"),
```
[the remark this answers is not in the document]
Comment (xterao) [reply src/main/kotlin/org/domaframework/doma/intellij/formatter/util/SqlKeywordUtil.kt:323]: There’s no inconsistency in the keyword group associations.
This is just handling combinations like PARTITION BY or WITHIN GROUP.
